src/rustup-cli/common.rs

```diff
@@ -402,11 +404,22 @@ pub fn list_overrides(cfg: &Cfg) -> Result<()> {
     Ok(())
 }
 
+git_testament!(TESTAMENT);
+
+lazy_static! {
+    static ref VERSION_STRING: String = {
+        let testament_string = format!("{}", TESTAMENT);
+        let pkg_version = env!("CARGO_PKG_VERSION");
+        if &testament_string[..pkg_version.len()] == pkg_version {
+            testament_string
+        } else {
+            format!("{} really {}", pkg_version, testament_string)
+        }
+    };
+}
```

Comment:
I think it would be fine to create the version string from scratch every time, rather than use a lazy static - it's not going to be on a hot path and lazy static is an extra dep and complicates the code

Reply:
The issue is that clap requires a `&str` passed to `.version()` and in `rustup_mode.rs` we return a clap `App` from a function which makes lifetimes a bit hard to deal with.